pkg/kn/commands/namespaced.go

```diff
@@ -31,19 +31,13 @@ func AddNamespaceFlags(flags *pflag.FlagSet, allowAll bool) {
 // GetNamespace returns namespace from command specified by flag
 func GetNamespace(cmd *cobra.Command) (string, error) {
 	namespace := cmd.Flag("namespace").Value.String()
-	if cmd.Flags().Lookup("all-namespaces") == nil {
-		if namespace == "" {
-			return defaultNamespace, nil
-		}
-		return namespace, nil
-	}
-
 	all, err := cmd.Flags().GetBool("all-namespaces")
```

Comment:
Well, this wont work for `service create` as it doesn't define the `all-namespaces` flag. My bad, updating the PR.

Reply:
fixed